tests/integration/test_overleaf.py::TestOverleaf::test_local

I found that for the remote test `TestOverleaf.test_local` the solution should be easy: after #467 we did not update our repository secrets, which are still `OVERLEAF_EMAIL` and `OVERLEAF_PASSWORD ` but there should be instead `OVERLEAF_TOKEN` 

@dfm I do not have enough permissions to fix this, because even if I can add a new repositpry secret I do not have the Overleaf account associated to this GitHub oganization (or if we were using @rodluger account)

Here is an example traceback from the remote workflow:

```python
___________________________ TestOverleaf.test_local ____________________________

self = <test_overleaf.TestOverleaf object at 0x7eff2a6db850>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
>           self.startup()

tests/integration/helpers/temp_repo.py:279: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/integration/test_overleaf.py:32: in startup
    overleaf.wipe_remote(self.overleaf_id)
../../../micromamba/envs/test-env/lib/python3.11/site-packages/showyourwork/overleaf.py:161: in wipe_remote
    overleaf_token = get_overleaf_credentials()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
```

overleaf_token = 'OVERLEAF_TOKEN', error_if_missing = False

    def get_overleaf_credentials(
        overleaf_token="OVERLEAF_TOKEN",
        error_if_missing=False,
    ):
        """
        Return the user's Overleaf token, stored in env vars.
    
        Args:
            overleaf_token (str, optional): Environment variable containing the
                Overleaf account password. Default ``OVERLEAF_TOKEN``.
            error_if_missing (bool, optional): Default ``False``.
    
        """
        val = os.getenv(overleaf_token, None)
        if val is None or not len(val):
            if error_if_missing:
                level = "error"
            else:
                # This exception is caught in the enclosing scope
                level = "warn"
>           raise exceptions.MissingOverleafCredentials(level=level)
E           showyourwork.exceptions.overleaf.MissingOverleafCredentials

../../../micromamba/envs/test-env/lib/python3.11/site-packages/showyourwork/overleaf.py:79: MissingOverleafCredentials
------------------------------ Captured log call -------------------------------
DEBUG    showyourwork:subproc.py:14 Initialized empty Git repository in /tmp/tmpxau7vvpm/.git/

WARNING  showyourwork:base.py:69 Overleaf credential `OVERLEAF_TOKEN` not found. This should be set as both environment variable and GitHub repository secret.
=============================== warnings summary ===============================
tests/integration/test_overleaf.py::TestOverleaf::test_local
  /home/runner/micromamba/envs/test-env/lib/python3.11/site-packages/showyourwork/exceptions/base.py:69: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
    get_logger().warn(message)

--- a/src/showyourwork/overleaf.py
+++ b/src/showyourwork/overleaf.py
@@ -165,43 +165,59 @@ def wipe_remote(project_id, tex=None):
             cwd=cwd,
             secrets=[overleaf_token],
         )
-        get_stdout(["git", "rm", "-r", "*"], cwd=cwd)
+
+        # Try to remove all files, but don't fail if there are no files to remove
+        def rm_callback(code, stdout, stderr):
+            if code != 0:
+                # Check if the error is because no files match the pattern
+                if (
+                    "non corrisponde ad alcun file" in stderr
+                    or "did not match any files" in stderr
+                ):
+                    # This is expected when the repo is empty, so we ignore it
+                    pass
+                else:
+                    # Re-raise other git rm errors
+                    raise exceptions.CalledProcessError(stderr)
+
+        get_stdout(["git", "rm", "-r", "*"], cwd=cwd, callback=rm_callback)
         with open(Path(cwd) / "main.tex", "w") as f:
             print(tex, file=f)
         get_stdout(["git", "add", "main.tex"], cwd=cwd)
 
-        def callback(code, stdout, stderr):
+        # Check if there are changes to commit using git status
+        def status_callback(code, stdout, stderr):
             if code != 0:
-                if (
-                    "Your branch is up to date" in stdout + stderr
-                    or "nothing to commit" in stdout + stderr
-                    or "nothing added to commit" in stdout + stderr
-                ):
-                    pass
-                else:
-                    raise exceptions.CalledProcessError(stdout + "\n" + stderr)
-            else:
-                get_stdout(
-                    ["git", "push", url, "master"],
-                    cwd=cwd,
-                    secrets=[overleaf_token],
-                    callback=check_for_rate_limit,
-                )
+                raise exceptions.CalledProcessError(stderr)
+            return stdout.strip()
 
-        get_stdout(
-            [
-                "git",
-                "-c",
-                "user.name='showyourwork'",
-                "-c",
-                "user.email='showyourwork'",
-                "commit",
-                "-am",
-                "automatic showyourwork update",
-            ],
+        status_output = get_stdout(
+            ["git", "status", "--porcelain"],
             cwd=cwd,
-            callback=callback,
+            callback=status_callback,
         )
+
+        if status_output:
+            # There are changes to commit and push
+            get_stdout(
+                [
+                    "git",
+                    "-c",
+                    "user.name='showyourwork'",
+                    "-c",
+                    "user.email='showyourwork'",
+                    "commit",
+                    "-am",
+                    "automatic showyourwork update",
+                ],
+                cwd=cwd,
+            )
+            get_stdout(
+                ["git", "push", url, "master"],
+                cwd=cwd,
+                secrets=[overleaf_token],
+                callback=check_for_rate_limit,
+            )
 
 
 def setup_remote(project_id, path=None):
@@ -376,48 +392,48 @@ def push_files(files, project_id, path=None):
         logger.warning(f"Skipping missing file(s): {skip_list}")
         files = list(set(files) - set(skip))
 
-    # Commit callback
-    def callback(code, stdout, stderr):
-        if stdout:
-            logger.debug(stdout)
-        file_list = " ".join([str(s) for s in files])
+    # Check if there are changes to commit using git status
+    def status_callback(code, stdout, stderr):
         if code != 0:
-            if (
-                "Your branch is up to date" in stdout + stderr
-                or "nothing to commit" in stdout + stderr
-                or "nothing added to commit" in stdout + stderr
-            ):
-                logger.warning(f"No changes to commit to Overleaf: {file_list}")
-            else:
-                raise exceptions.CalledProcessError(stdout + "\n" + stderr)
-        else:
-            logger.info(f"Pushing changes to Overleaf: {file_list}")
+            raise exceptions.CalledProcessError(stderr)
+        return stdout.strip()
 
-    # Commit!
-    get_stdout(
-        [
-            "git",
-            "-c",
-            "user.name='showyourwork'",
-            "-c",
-            "user.email='showyourwork'",
-            "commit",
-            "-m",
-            "[showyourwork] automatic showyourwork update",
-        ],
+    status_output = get_stdout(
+        ["git", "status", "--porcelain"],
         cwd=str(paths.user(path=path).overleaf),
-        callback=callback,
+        callback=status_callback,
     )
 
-    # Push (again being careful about secrets)
-    overleaf_token = get_overleaf_credentials()
-    url = f"https://git:{overleaf_token}@git.overleaf.com/{project_id}"
-    get_stdout(
-        ["git", "push", url, "master"],
-        cwd=str(paths.user(path=path).overleaf),
-        secrets=[overleaf_token],
-        callback=check_for_rate_limit,
-    )
+    file_list = " ".join([str(s) for s in files])
+    if status_output:
+        # There are changes to commit
+        get_stdout(
+            [
+                "git",
+                "-c",
+                "user.name='showyourwork'",
+                "-c",
+                "user.email='showyourwork'",
+                "commit",
+                "-m",
+                "[showyourwork] automatic showyourwork update",
+            ],
+            cwd=str(paths.user(path=path).overleaf),
+        )
+        logger.info(f"Pushing changes to Overleaf: {file_list}")
+
+        # Push (again being careful about secrets)
+        overleaf_token = get_overleaf_credentials()
+        url = f"https://git:{overleaf_token}@git.overleaf.com/{project_id}"
+        get_stdout(
+            ["git", "push", url, "master"],
+            cwd=str(paths.user(path=path).overleaf),
+            secrets=[overleaf_token],
+            callback=check_for_rate_limit,
+        )
+    else:
+        # No changes to commit
+        logger.warning(f"No changes to commit to Overleaf: {file_list}")
 
 
 def pull_files(
@@ -558,35 +574,37 @@ def pull_files(
             )
 
     if commit_changes:
+        # Check if there are any changes to commit using git status
+        def status_callback(code, stdout, stderr):
+            if code != 0:
+                raise exceptions.CalledProcessError(stderr)
+            return stdout.strip()
 
-        def callback(code, stdout, stderr):
-            if code == 0:
-                logger.info(
-                    "Overleaf changes committed to the repo. Don't forget to push!"
-                )
-            elif (
-                "Your branch is up to date" in stdout + stderr
-                or "nothing to commit" in stdout + stderr
-                or "nothing added to commit" in stdout + stderr
-            ):
-                logger.warning("No Overleaf changes to commit to the repo.")
-            else:
-                raise exceptions.CalledProcessError(stdout + "\n" + stderr)
-
-        get_stdout(
-            [
-                "git",
-                "-c",
-                "user.name='showyourwork'",
-                "-c",
-                "user.email='showyourwork'",
-                "commit",
-                "-m",
-                "[showyourwork] overleaf sync",
-            ],
+        status_output = get_stdout(
+            ["git", "status", "--porcelain"],
             cwd=paths.user(path=path).repo,
-            callback=callback,
+            callback=status_callback,
         )
+
+        if status_output:
+            # There are changes to commit
+            get_stdout(
+                [
+                    "git",
+                    "-c",
+                    "user.name='showyourwork'",
+                    "-c",
+                    "user.email='showyourwork'",
+                    "commit",
+                    "-m",
+                    "[showyourwork] overleaf sync",
+                ],
+                cwd=paths.user(path=path).repo,
+            )
+            logger.info("Overleaf changes committed to the repo. Don't forget to push!")
+        else:
+            # No changes to commit
+            logger.warning("No Overleaf changes to commit to the repo.")
 
         if push_changes:
             get_stdout(["git", "push"], cwd=paths.user(path=path).repo)

--- a/tests/integration/test_overleaf.py
+++ b/tests/integration/test_overleaf.py
@@ -19,7 +19,7 @@ class TestOverleaf(TemporaryShowyourworkRepository, ShowyourworkRepositoryAction
 
     # No need to test this on CI
     local_build_only = True
-    overleaf_id = "6409f16f438b5fb7c4dfa837"
+    overleaf_id = "68cd54011c99150d9045d2b6"
 
     # Overleaf rate limit error re-try settings
     auth_retries = 1

